mne/viz/topo.py

```diff
@@ -855,23 +855,27 @@ def plot_topo_image_epochs(epochs, layout=None, sigma=0., vmin=None,
         Figure distributing one image per channel across sensor topography.
     """
     scalings = _handle_default('scalings', scalings)
-    data = epochs.get_data()
-    scale_coeffs = list()
-    for idx in range(epochs.info['nchan']):
-        ch_type = channel_type(epochs.info, idx)
-        scale_coeffs.append(scalings.get(ch_type, 1))
-    vmin, vmax = _setup_vmin_vmax(data, vmin, vmax)
 
     if layout is None:
         layout = find_layout(epochs.info)
+    epochs_ = epochs.copy().pick_channels(layout.names)
+    data = epochs_.get_data()
```

Comment:
In some cases this might be much slower than indexing into the data array, right?

Reply:
Rebase and do `epochs.get_data(picks=picks)` instead.